File: idea/Aplicacion/src/Logica/Usuario.java
package Logica;

import java.util.ArrayList;
import java.util.Scanner;

//Descrive al usuario del programa
class Usuario {
	private String nombre;
	private String email;
	private String nombreCompleto;
	private String logoNegocio;
	private String descripcionNegocio;
	private String terminosNegocio;
	private Notificacion notificacion;
	private ContactoUsuario contactoUsuario;
	private ArrayList<ContactosLocales> contactosLocales;
	private ArrayList<ContactosPendientes> contactosPendientes;

	//Genera al usuario
	Usuario(String nombre, String email, String nombreCompleto, String logoNegocio, String descripcionNegocio,
			String terminosNegocio) {
		this.nombre = nombre;
		this.email = email;
		this.nombreCompleto = nombreCompleto;
		this.logoNegocio = logoNegocio;
		this.descripcionNegocio = descripcionNegocio;
		this.terminosNegocio = terminosNegocio;
		//la siguite linea necesita modificacion
		this.notificacion = new Notificacion();
		//Crea su propio contacto
		this.contactoUsuario = new ContactoUsuario(this);
		this.contactosLocales = new ArrayList<ContactosLocales>();
		this.contactosPendientes = new ArrayList<ContactosPendientes>();
	}

	//Muestra el estado del objeto usuario
	@Override
	public String toString() {
		return "Usuario [nombre=" + nombre + ", email=" + email + ", nombreCompleto=" + nombreCompleto
				+ ", logoNegocio=" + logoNegocio + ", descripcionNegocio=" + descripcionNegocio + ", terminosNegocio="
				+ terminosNegocio + ", notificacion=" + notificacion + ", contactoUsuario=" + contactoUsuario
				+ ", contactosLocales=" + contactosLocales + ", contactosPendientes=" + contactosPendientes + "]";
	}
	
	//Cambia los atributos del usuario, menos notificaciones y contactos
	//Resive al unico escaner del programa como parametro.
	void editarPerfil(Scanner sc) {
		int opcion;
		do {
			System.out.println(this);
			System.out.println("Cambiar:\n0 Cancela\n1 nombre\n2 email\n3 nombreCompleto\n4 logoNegocio\n5 descripcionNegocio"
					+ "\n6 terminosNegocio");
			//Resive un entero (no elimina \n).
			opcion = sc.nextInt();
			//Elimina el \n
			sc.nextLine();
			
			switch (opcion) {
				case 0:
					System.out.println("Termino modificacion de Perfil");
					break;
					
				case 1:
					System.out.println("Nombre actual: " + getNombre());
					System.out.print("Nuevo nombre: ");
					//Resive el nombre (Puede ser mas de una palabra y elimina \n).
					setNombre(sc.nextLine());
					break;
					
				case 2:
					System.out.println("Email actual: " + getEmail());
					System.out.print("Nuevo Email: ");
					//Resive el email (una palabra y no elimina \n).
					setEmail(sc.next());
					//Elimina el \n
					sc.nextLine();
					break;
					
				case 3:
					System.out.println("Nombre completo actual: " + getNombreCompleto());
					System.out.print("Nuevo nombre completo: ");
					//Resive el nombre completo (Puede ser mas de una palabra y elimina \n).
					setNombreCompleto(sc.nextLine());
					break;
					
				case 4:
					System.out.println("Logo de negocio actual: " + getLogoNegocio());
					System.out.print("Nuevo logo de negocio: ");
					//Resive el logo (Puede ser mas de una palabra y elimina \n).
					setLogoNegocio(sc.nextLine());
					break;
					
				case 5:
					System.out.println("Descripcion de negocio actual: " + getDescripcionNegocio());
					System.out.print("Nueva descripcion de negocio: ");
					//Resive la descripcion (Puede ser mas de una palabra y elimina \n).
					setDescripcionNegocio(sc.nextLine());
					break;
					
				case 6:
					System.out.println("Terminos de negocio actual: " + getTerminosNegocio());
					System.out.print("Nuevo terminos de negocio: ");
					//Resive los terminos (Puede ser mas de una palabra y elimina \n).
					setTerminosNegocio(sc.nextLine());
					break;
					
				default:
					System.out.println("Seleccione un numero entre 0 y 6");
					break;	
			}
		}while (opcion != 0);
		
	}
	
	//Añade un nuevo contacto al array de contactos pendites
	//Resive al unico escaner del programa como parametro.
	void crearNuevoContacto(Scanner sc) {
		//Llama al metodo de clase crear contacto de pendientes que devuelve un contacto 
		contactosPendientes.add(ContactosPendientes.crearContactoPendiente(sc));
	}
	
	//funcionalidad sin terminar
	//Resive al unico escaner del programa como parametro.
	void empezarChat(Scanner sc) {
		int opcion;
		do {
			System.out.println("Elige un contacto");
			System.out.println("0 cancelar");
			for (int i = 0; i < contactosPendientes.size(); i++) {
				System.out.println((i+1) + " " + contactosPendientes.get(i).toString());
			}
			opcion = sc.nextInt();
			sc.nextLine();
			
			if ((opcion > 0) && (opcion <= contactosPendientes.size())) {
				System.out.println(contactosPendientes.get(opcion - 1).getEmail());
				
			} else if (opcion != 0) {
				System.out.println("Seleccione un numero entre 0 y " + contactosPendientes.size());
				
			} else {
				System.out.println("Termino empezar chat");
				
			}
			
		} while (opcion != 0);
		
	}
	
	String getNombre() {
		return nombre;
	}

	void setNombre(String nombre) {
		this.nombre = nombre;
	}

	String getEmail() {
		return email;
	}

	void setEmail(String email) {
		this.email = email;
	}

	String getNombreCompleto() {
		return nombreCompleto;
	}

	void setNombreCompleto(String nombreCompleto) {
		this.nombreCompleto = nombreCompleto;
	}

	String getLogoNegocio() {
		return logoNegocio;
	}

	void setLogoNegocio(String logoNegocio) {
		this.logoNegocio = logoNegocio;
	}

	String getDescripcionNegocio() {
		return descripcionNegocio;
	}

	void setDescripcionNegocio(String descripcionNegocio) {
		this.descripcionNegocio = descripcionNegocio;
	}

	String getTerminosNegocio() {
		return terminosNegocio;
	}

	void setTerminosNegocio(String terminosNegocio) {
		this.terminosNegocio = terminosNegocio;
	}

	Notificacion getNotificacion() {
		return notificacion;
	}

	void setNotificacion(Notificacion notificacion) {
		this.notificacion = notificacion;
	}

	ContactoUsuario getContactoUsuario() {
		return contactoUsuario;
	}

	void setContactoUsuario(ContactoUsuario contactoUsuario) {
		this.contactoUsuario = contactoUsuario;
	}

	ArrayList<ContactosLocales> getContactosLocales() {
		return contactosLocales;
	}

	void setContactosLocales(ArrayList<ContactosLocales> contactosLocales) {
		this.contactosLocales = contactosLocales;
	}

	ArrayList<ContactosPendientes> getContactosPendientes() {
		return contactosPendientes;
	}

	void setContactosPendientes(ArrayList<ContactosPendientes> contactosPendientes) {
		this.contactosPendientes = contactosPendientes;
	}
	
}

File: Python/uiMain/framesFuncionalidades/usuario.py
from contactoUsuario import ContactoUsuario
from tarjetaNegocioUsuario import TarjetaNegocioUsuario
from contactosLocales import ContactosLocales
from contactosPendientes import ContactosPendientes
from mensajeNegocio import MensajeNegocio
from MensajeSocial import MensajeSocial

#Describe al usuario del programa
class Usuario:
    _contactoUsuario: ContactoUsuario
    _tarjetaNegocioUsuario: TarjetaNegocioUsuario
    _contactosLocales: list[ContactosLocales]
    _contactosPendientes: list[ContactosPendientes]

    #Genera al Usuario y crea su ContactoUsuario
    def __init__(self, nombre="Nombre",email="Email",nombreCompleto="Nombre completo",logoNegocio="logo",descripcionNegocio="descripcion",terminosNegocio="terminos") -> None:
        self._contactoUsuario = ContactoUsuario(nombre, email, nombreCompleto, logoNegocio, descripcionNegocio, terminosNegocio)
        self._contactosLocales = [],
        self._contactosPendientes = []

    #Añade un nuevo contacto al array de contactos pendites
    def crearNuevoContacto(self, nombre: str, email: str, nombreCompleto: str) -> None:
        #Llama al metodo de clase crear contacto de pendientes que devuelve un contacto
        self._contactosPendientes.append(ContactosPendientes(nombre, email, nombreCompleto))

    #Depende del metodo empezarChat para optener el indice del contacto
	#vuele un contacto pendiente en local
    def volverContactoLocal(self, indice: int) -> None: 
        contactosPendientes: ContactosPendientes = self._contactosPendientes[indice]
        self._contactosLocales = ContactosLocales(contactosPendientes.getNombre(), contactosPendientes.getEmail(), contactosPendientes.getNombreCompleto())
        self._contactosPendientes.remove(contactosPendientes)

    #le agrega un mensaje de negocios al contacto local recien creado
    def anadirMensajeNegocio(self, titulo: str, cuerpo: str) -> None:
        ultimoContactoCreado: ContactosLocales = self._contactosLocales[self._contactosLocales.count() - 1]
        ultimoContactoCreado.getMensajes().append(
            MensajeNegocio(
                titulo, 
                cuerpo, 
                self._contactoUsuario.getLogoNegocio(),
                self._contactoUsuario.getDescripcionNegocio(),
                self._contactoUsuario.getTerminosNegocio(),
                self._contactoUsuario.getNombreCompleto(),
                ultimoContactoCreado.getNombreCompleto()
                )
        )

    #le agrega un mensaje social al contacto local recien creado
    def anadirMensajeSocial(self, titulo: str, cuerpo: str) -> None:
        ultimoContactoCreado: ContactosLocales = self._contactosLocales[self._contactosLocales.count() - 1]
        ultimoContactoCreado.getMensajes().append(
            MensajeSocial(
                titulo, 
                cuerpo, 
                self._contactoUsuario.getNombreCompleto(),
                ultimoContactoCreado.getNombreCompleto()
                )
        )
    
    #
    def mostrarChats(self) -> str:
        texto: str = ""
        for contactoLocal in self._contactosLocales:
            texto += "email: " + contactoLocal.getEmail() + "\n"
            texto += "Mensajes: " + "\n"
            for mensajes in contactoLocal.getMensajes():
                texto += mensajes + "\n"

        return texto

    #Devuelve en contacto pendiente en la pocision i
    def getContactoPendiente(self, indice: int) -> ContactosPendientes:
        return self._contactosPendientes[indice]

    #Devuelve en contacto local en la pocision i
    def getContactoLocal(self, indice: int) -> ContactosLocales:
        return self._contactosLocales[indice]

    def getContactoUsuario(self) -> ContactoUsuario:
        return self._contactoUsuario

    def setContactoUsuario(self, contactoUsuario: ContactoUsuario) -> None:
        self._contactoUsuario = contactoUsuario

    def getContactosLocales(self) -> list[ContactosLocales]:
        return self._contactosLocales

    def setContactosLocales(self, contactosLocales: ContactosLocales) -> None:
        self._contactosLocales = contactosLocales

    def getContactosPendientes(self) -> list[ContactosPendientes]:
        return self._contactosPendientes

    def setContactosPendientes(self, contactosPendientes: ContactosPendientes) -> None:
        self._contactosPendientes = contactosPendientes

    def getTarjetaNegocio(self) -> TarjetaNegocioUsuario:
        return self._tarjetaNegocioUsuario
    
    def setTarjetaNegocioUsuario(self, tarjetaNegocioUsuario: TarjetaNegocioUsuario) -> None:
        self._tarjetaNegocioUsuario = tarjetaNegocioUsuario

    def __str__(self) -> str:
        return "Usuario {" + "contactoUsuario=" + self.getContactoUsuario() + ",\ntarjetaNegocioUsuario=" + self.getTarjetaNegocio().__str__() + ",\ncontactosLocales=" + self.getContactosLocales().__str__() +",\ncontactosPendientes=" + self.getContactosPendientes().__str__() + '\n}'



            


    
